lib/Parse/ParseDecl.cpp

```diff
@@ -892,6 +892,17 @@ bool Parser::parseDifferentiationParametersClause(
             paramLoc, paramName));
         break;
       }
+      case tok::integer_literal: {
+        unsigned int paramNum;
```

Comment:
```suggestion
        unsigned paramNum;
```

This LGTM, though it is more common in the LLVM/Swift code base to use `unsigned` directly.

Reply:
Ahh yup! Actually wouldn't `size_t` be better here given that this has to with the number of parameters aka size of the array? https://en.wikipedia.org/wiki/C_data_types#stddef.h